MetaMask/metamask-extension
# Ability to buy tokens with Coinbase Pay and Transak

## Explanation

Current State: Only native currencies can be bought using on ramp through the buy button
After Change: Ability to purchase tokens defined in `BUYABLE_CHAINS_MAP` using Coinbase Pay and Transak

Changes:
- split modals differentiating between native currencies and tokens, similar strategy used for `eth-overview` and `token-overview`
- if a token is defined as buyable on the `BUYABLE_CHAINS_MAP` on at least one provider, a buy button will show on the token overview page


## More Information



See ONRAMP-189 and ONRAMP-190

## Screenshots/Screencaps



### Before
![Screenshot 2022-08-11 at 13 24 42](https://user-images.githubusercontent.com/8658960/184123294-ca248deb-5fb3-4fce-b932-b1c03647c540.png)



### After
![Screenshot 2022-08-11 at 13 24 57](https://user-images.githubusercontent.com/8658960/184123312-f530a3ef-2fa3-413d-aafc-e53dd2007a46.png)
![Screenshot 2022-08-11 at 13 25 16](https://user-images.githubusercontent.com/8658960/184123314-0fb8cca9-3923-4ce9-ae05-dfbe7a09445f.png)



## Manual Testing Steps

These changes apply only to Ethereum mainnet

- Import a token
- If token is on the [Transak list](https://github.com/nikoferro/metamask-extension/commit/66d8717ecdecf60af3dc6fb8d97a813372ae0210#diff-4e682f9ffe2eed5743a5d0b761ffdab503afaad448954bd69ce2f7c8bca000d1R266) or on the [Coinbase Pay list](https://github.com/nikoferro/metamask-extension/commit/66d8717ecdecf60af3dc6fb8d97a813372ae0210#diff-4e682f9ffe2eed5743a5d0b761ffdab503afaad448954bd69ce2f7c8bca000d1R309), it should display a "Buy" button
- Depending on which list it shows, it should give you the option to buy the token on said provider
- Once you click on the provider's "Continue to" button, a window on said provider should open, and the token to buy should already be preselected



## Pre-Merge Checklist


### + If there are functional changes:
